### PR TITLE
#8 Add support for HTTP headers

### PR DESCRIPTION
**Pull request by albx79**
Add a new parameter that allows users to specify custom http headers.

### CODE REVIEW

File: src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java [hunk 1]
```diff
@@ -230,6 +237,12 @@
     @Parameter(property = "checkSignature", defaultValue = "false")
     private boolean checkSignature;
 
+    /**
+     * A list of additional HTTP headers to send with the request
+     */
+    @Parameter(property = "download.plugin.headers")
+    private List<String> headers = new ArrayList<>();
```
Comment (longtimeago) [inline src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java:244]: @albx79 would it make sense to have a Map here?

**Comment by longtimeago** [comment]
@albx79 thank you for the PR! Could you, please, add an integration test?

**Comment by albx79** [comment]
Any idea how I could do it? I'd have to find a website that requires a
custom header to work.

Il giorno mer 29 apr 2020 alle ore 09:58 Paul Polishchuk <
notifications@github.com> ha scritto:

**Comment by albx79** [comment]
hmm maybe I found a solution

Il giorno mer 29 apr 2020 alle ore 11:42 alberto colombo <albx79@gmail.com>
ha scritto:

File: src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java (hunk 1 above)
Comment (albx79) [reply src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java:244]: I don't know, compared to a simple "header-name: value", I think the map is just adding syntactic noise. Apache Http Client provides a class to parse the colon-separated string, so I don't thing a map would buy us much.

Comment (longtimeago) [reply src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java:244]: @albx79 the main benefit of a Map is that we don't need to parse anything at all. The only thing we would need it construct `Header` instances from K/V pairs.
Config example
```xml
<headers>
	<X-Custom-1>first custom header</X-Custom-1>
	<X-Custom-2>second custom header</X-Custom-2>
</headers>
```
And `getAdditionalHeaders()` becomes as simple as this
```java
        return this.headers.entrySet().stream().map(
            pair -> new BasicHeader(pair.getKey(), pair.getValue())
        ).collect(Collectors.toList());
```
Make sense?

**Comment by albx79** [comment]
I've turned the list into a map. Yeah removing the parsing code made it much cleaner.

Do you think you can merge it now?

Thanks

**Comment by albx79** [comment]
I hate to sound pushy, but can you do a final review to my PR and, if it's good, merge it? As I mentioned in #163 , we have a project blocked on this.

**Pull request merged**